tests/test_watch.py

```diff
@@ -104,11 +104,38 @@ def test_paginate_helper(self):
         size = 10
         page = 0
         total = len(logs)
-        paginated_logs, pages = paginate(
+        paginated_logs, pages, page = paginate(
             self.user.get_recent_log_ids(), total, page, size)
         page_num = math.ceil(total / float(size))
         assert_equal(len(list(paginated_logs)), total)
         assert_equal(page_num, pages)
 
+    def test_paginate_not_go_to_negative(self):
+        self._watch_project(self.project)
+        logs = list(self.user.get_recent_log_ids())
+        size = 10
+        page = -1
+        total = len(logs)
+        paginated_logs, pages, page = paginate(
+            self.user.get_recent_log_ids(), total, page, size)
+        page_num = math.ceil(total / float(size))
+        assert_equal(len(list(paginated_logs)), total)
+        assert_equal(page_num, pages)
+        assert_equal(page, 0)
```

Comment:
This seems to be incorrect behavior.  If the user requests page `-1`, they shouldn't get back page `0` - they should get an `HTTP 404`, as that page doesn't exist.

Likewise, if there are 3 pages total and the user requests page `4`, they should also get back an `HTTP 404`.